machine_learning/similarity_search.py

```diff
@@ -0,0 +1,113 @@
+"""
+Simularity search is a search algorithm for finding the nearest vector from
+vectors, used in natural language processing.
+In this algorithm, it calculates distance with euclidean distance and
+returns a list containing two data for each vector:
+    1. the nearest vector
+    2. distance between the vector and the nearest vector
+"""
+import math
+from typing import Union
+
+import numpy as np
+
+InputVal = Union[int, float, np.ndarray]
+
+
+def euclidean(input_a: InputVal, input_b: InputVal):
+    """
+    Calculates euclidean distance between two data. The result should be float.
+    >>> euclidean(0, 1)
+    1.0
+    >>> euclidean(np.array([0, 1]), np.array([1, 1]))
+    1.0
+    >>> euclidean(np.array([0, 0, 0]), np.array([0, 0, 1]))
+    1.0
+    """
+    dist = 0
+
+    if type(input_a) == type(input_b):
```

Comment:
Please read PEP8 to discover why directly comparing types is not recommended.

I would recommend doing this as a try / except block so this function can accept lists, tuples, and other iterators.

Reply:
Thinking of the usage of euclidean, it will only get numpy.ndarrays.

But also blocked it with try / except just in case. Thanks!